Shared/Samples/Add building scene layer/AddBuildingSceneLayerView.swift

```diff
@@ -0,0 +1,106 @@
+    @State private var scene: ArcGIS.Scene = {
+        let scene = Scene(
+            basemapStyle: .arcGISTopographic,
+            viewingMode: .local
+        )
+        
+        // Adds a surface.
+        
+        let elevationSource = ArcGISTiledElevationSource(
+            url: URL(string: "https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer")!
+        )
+        scene.baseSurface.addElevationSource(elevationSource)
+        
+        return scene
+    }()
+    
+    /// A Boolean value that indicates if the full model of the building scene layer is showing or not.
+    @State private var showsFullModel = false
+    
+    /// The camera for zooming the local scene view to the building scene layer.
+    @State private var camera: Camera? = Camera(
+        location: Point(x: -13_045_109, y: 4_036_614, z: 511, spatialReference: .webMercator),
+        heading: 343,
+        pitch: 64,
+        roll: 0
+    )
+    
+    /// The overview sublayer which represents the exterior shell of the building.
+    @State private var overviewSublayer: BuildingSublayer?
+    
+    /// The full model sublayer which contains all the features of the building.
+    @State private var fullModelSublayer: BuildingSublayer?
+    
+    var body: some View {
+        LocalSceneView(scene: scene, camera: $camera)
+            .task {
+                let buildingSceneLayer = BuildingSceneLayer(
+                    url: URL(string: "https://3dcities.maps.arcgis.com/home/item.html?id=ebae5f766aac41ba9ec588f66028a6a9")!
+                )
+                
+                // Sets the altitude offset of the building scene layer.
+                // Upon first inspection of the model, it does not line up with the global
+                // elevation layer perfectly. To fix this, add an altitude offset to align
+                // the model with the ground surface.
+                buildingSceneLayer.altitudeOffset = 1
+                
+                try? await buildingSceneLayer.load()
+                
+                // Adds building scene layer to scene.
+                scene.addOperationalLayer(buildingSceneLayer)
+                
+                // Get the overview and full model sublayers for the
+                // toggle.
+                let sublayers = buildingSceneLayer.sublayers
+                overviewSublayer = sublayers.first(where: { $0.name == "Overview" })
+                fullModelSublayer = sublayers.first(where: { $0.name == "Full Model" })
+            }
+            .toolbar {
+                ToolbarItem(placement: .bottomBar) {
+                    // Shows the toggle when we have a valid full model
+                    // sublayer since we need that for the toggle.
+                    if fullModelSublayer != nil {
+                        Toggle("Full Model", isOn: $showsFullModel)
+                            .toggleStyle(.button)
```

Comment:
Due to #321 we typically avoid toggle button to better support Mac Catalyst.

@chriswebb09 can you share what are the alternatives?

Reply:
I added the solution above.